### ISSUE TITLE
[Northstar UX] Remove header <h> tags from markdown rendering

### ISSUE THREAD

**Issue by ezraeinhorn624** [civiform/civiform#10213]
**Describe the bug**
[Currently](https://github.com/civiform/civiform/blob/5d9d66d9a24eefc4e279d0b17c7871f2bb3931e6/server/app/views/components/TextFormatter.java#L202), we only remove `h1` tags from markdown for a11y reasons. We should also strip all other `h` tags from markdown to preserve the accessibility hierarchy of the page.

See [this issue](https://github.com/civiform/civiform/blob/5d9d66d9a24eefc4e279d0b17c7871f2bb3931e6/server/app/views/components/TextFormatter.java#L202) and [this Slack thread](https://civiform.slack.com/archives/C06H11X3ERE/p1743692019079219) for more details.

**To Reproduce**

1. Go to any page that has rendered markdown, for e.g. [pre-northstar programs (applicant)](https://staging-aws.civiform.dev/programs) or [post-northstar program overview (applicant)](https://staging-aws.civiform.dev/programs/comprehensive-sample-program)
2. Enter dev mode
3. Inspect the rendered markdown to view `h` tags

**Expected behavior**
Headings should be converted to `p` or `span` tags, and possibly bolded.

**Comment by dkatzz** [comment]
> [@shreyachatterjee00](https://github.com/shreyachatterjee00) Could you confirm that CEs are okay with this?

We're moving this to fast follows, but we should add this to the northstar checklist for CEs that they should ensure they don't have any headings or sub-headings before they migrate and if they do, we can work with them to choose other markdown formatting or fix this before they turn on northstar.

**Comment by shreyachatterjee00** [comment]
@dkatzz @melanie-exygy actually turns out Charlotte isn't using headings anywhere except in text that will be put in the program overview anyway. so we're good to do this whenever!

**Comment by melanie-exygy** [comment]
> Should stripping the h tags only apply to North Star? I think ideally it would apply to both, but sounds like that might break some legacy formatting.

Yeah, I don't think it's worth putting much effort into pre-NS since the North Star launch is imminent.  It sounds like several governments will be moving headings that currently exist in markdown into the program overview page.  If we want to fix this in legacy, we'd have to find another place to move the headings to.
